tools/nodejs_api/src_cpp/include/node_connection.h

```diff
@@ -13,5 +13,5 @@ class NodeConnection : public Napi::ObjectWrap<NodeConnection> {
   Napi::Value Execute(const Napi::CallbackInfo& info);
   void SetMaxNumThreadForExec(const Napi::CallbackInfo& info);
   Napi::Value GetNodePropertyNames(const Napi::CallbackInfo& info);
-  unique_ptr<kuzu::main::Connection> connection;
+  shared_ptr<kuzu::main::Connection> connection;
```

Comment:
Why do you need a `shared_ptr` here? Why `unique_ptr` would not work?

Reply:
Because I'm passing the connection to the ExecuteWorker and still need the connection class itself to have access to connection for future executions.